homeassistant/components/agent_dvr/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""Support for Agent."""
+import asyncio
+import logging
+
+from agent import AgentError
+from agent.a import Agent
+
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONNECTION, DOMAIN as AGENT_DOMAIN, SERVER_URL
+
+ATTRIBUTION = "ispyconnect.com"
+DEFAULT_BRAND = "Agent DVR by ispyconnect.com"
+
+_LOGGER = logging.getLogger(__name__)
+
+FORWARDS = ["camera"]
+
+
+async def async_setup(hass, config):
+    """Old way to set up integrations."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up the Agent component."""
+    hass.data.setdefault(AGENT_DOMAIN, {})
+
+    server_origin = config_entry.data[SERVER_URL]
+
+    agent_client = Agent(server_origin, async_get_clientsession(hass))
+    try:
+        await agent_client.update()
+    except AgentError:
+        await agent_client.close()
+        raise ConfigEntryNotReady
+
+    if not agent_client.is_available:
+        raise ConfigEntryNotReady
+
+    agent_client.deviceList = await agent_client.get_devices()
```

Comment:
Why do we store the devices in `agent_client.deviceList`? They are already stored in `agent_client.devices`:
https://github.com/ispysoftware/agent-py/blob/4fd34278f3a8ba22e8894461ea0886e36943878e/agent/a.py#L37

https://github.com/ispysoftware/agent-py/blob/4fd34278f3a8ba22e8894461ea0886e36943878e/agent/a.py#L104-L111

```suggestion
    await agent_client.get_devices()
```